### ISSUE TITLE
User destruction broken for non-leaders

### ISSUE THREAD

**Issue by esoterik** [YaleSTC/vesta#410]
changing the membership destruction test to destroy a non leader:
```ruby
  it 'destroys a dependent membership on destruction' do
    user = FactoryGirl.create(:student, intent: 'on_campus')
    g = FactoryGirl.create(:drawless_group, size: 2)
    m_id = Membership.create(group: g, user: user).id
    user.destroy
    expect { Membership.find(m_id) }.to \
      raise_error(ActiveRecord::RecordNotFound)
  end
```
results in this error
```
  1) User destroys a dependent membership on destruction
     Failure/Error: user.destroy

     ActiveRecord::StatementInvalid:
       PG::UndefinedColumn: ERROR:  column groups.user_id does not exist
       LINE 1: SELECT  "groups".* FROM "groups" WHERE "groups"."user_id" = ...
                                                      ^
       : SELECT  "groups".* FROM "groups" WHERE "groups"."user_id" = $1 LIMIT $2
     # ./spec/models/user_spec.rb:47:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # PG::UndefinedColumn:
     #   ERROR:  column groups.user_id does not exist
     #   LINE 1: SELECT  "groups".* FROM "groups" WHERE "groups"."user_id" = ...
     #                                                  ^
     #   ./spec/models/user_spec.rb:47:in `block (2 levels) in <top (required)>'
```

**Comment by orenyk** [comment]
This is fixed in #397, closing.

**Comment by orenyk** [comment]
Actually, reopening and will update the PR description.
